yt/visualization/profile_plotter.py

```diff
@@ -711,6 +727,64 @@ def _get_field_title(self, field_y, profile):
 
         return (x_title, y_title)
 
+    @invalidate_plot
+    def annotate_title(self, title):
+        r"""Set a title for the plot.
+
+        Parameters
+        ----------
+        title : str
+          The title to add.
+
+        Examples
+        --------
+
+        >>> plot.annotate_title("This is a particle plot")
+
+        """
+        self.plot_title = title
+        return self
+
+    def annotate_text(self, xpos=0.0, ypos=0.0, text=None, **text_kwargs):
+        r"""Allow the user to insert text onto the plot
+
+        The x-position and y-position must be given as well as the text string.
+        Add *text* tp plot at location *xpos*, *ypos* in plot coordinates
+        (see example below).
+
+        Parameters
+        ----------
+        field: str or tuple
+          The name of the field to add text to.
+        xpos: float
+          Position on plot in x-coordinates.
+        ypos: float
+          Position on plot in y-coordinates.
+        text: str
+          The text to insert onto the plot.
+        text_kwargs: dict
+          Dictionary of text keyword arguments to be passed to matplotlib
+
+        >>>  import yt
+        >>>  from yt.units import kpc
+        >>>  ds = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')
+        >>>  my_galaxy = ds.disk(ds.domain_center, [0.0, 0.0, 1.0], 10*kpc, 3*kpc)
+        >>>  plot = yt.ProfilePlot(my_galaxy, "density", ["temperature"])
+        >>>  plot.annotate_text(1e-26, 1e5, "This is annotated text in the plot area.")
+        >>>  plot.save()
+
+        """
+        for f in list(self.axes.keys()):
+            if self.plots[f].figure is not None and text is not None:
+                self.plots[f].axes.text(xpos, ypos, text,
+                                        fontproperties=self._font_properties,
+                                        **text_kwargs)
```

Comment:
You can delete this if statement if you decorate this function with `@invalidate_plot`.